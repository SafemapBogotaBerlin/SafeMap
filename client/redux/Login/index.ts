import { createSlice } from '@reduxjs/toolkit';

const loginSlice = createSlice({
  name: 'loggedin',
  initialState: {
    value: false,
  },
  reducers: {
    login: (state) => {
      state.value = !state.value;
    },
  },
});

export const { login } = loginSlice.actions;
export default loginSlice.reducer;
