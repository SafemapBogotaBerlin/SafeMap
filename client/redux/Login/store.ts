import { configureStore } from '@reduxjs/toolkit';
import loginReducer from '.';
export const store = configureStore({
  reducer: {
    test: loginReducer,
  },
});
export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
