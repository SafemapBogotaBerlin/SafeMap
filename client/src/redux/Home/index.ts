import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import type { PayloadAction } from '@reduxjs/toolkit';
import { Coordinates, Point, DataObject } from '../../../types/point';


interface HomeState {
  hotpoints: Coordinates[];
  data: DataObject;
  selectedPoint: Coordinates | null;
  status: 'idle' | 'pending' | 'succeeded' | 'rejected';
  error: string | null;
}
 

let data: DataObject = {};


const initialState: HomeState = {

  // this should be replaced by the call to the real time database
  hotpoints: [
    { latitude: 19.532608, longitude: -99.53209 },
    { latitude: 19.632608, longitude: -99.73209 },
    { latitude: 19.732608, longitude: -99.03 },
    {latitude: 4.820954117734985, longitude: -74.06501975735756}
  ],

  data: data,

  selectedPoint: null,
  status: 'idle',
  error: null,
};

const homeSlice = createSlice({
  name: 'home',
  initialState: initialState,
  reducers: {
    // updatePoints: (state) => {
    //   update a point might be needed later
    // },
    selectPoint: (state, action: PayloadAction<Coordinates>) => {
      state.selectedPoint = { ...action.payload };
    },
  },
});

export const { selectPoint } = homeSlice.actions;
export default homeSlice.reducer;
