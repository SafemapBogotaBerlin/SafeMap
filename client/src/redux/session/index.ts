import { createSlice } from '@reduxjs/toolkit';
import { UserData } from '../../types/index';

interface LoginState {
  authenticated: boolean;
  userData: UserData;
}

const initialState: LoginState = {
  authenticated: false,
  userData: null,
};

const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: {
    setUserData: (state, action) => {
      state.userData = action.payload;
    },

    authenticate: (state) => {
      state.authenticated = true;
    },

    updateUserName: (state, action) => {
      state.userData.name = action.payload;
    },

    logout: (state) => {
      state.authenticated = false;
      state.userData = null;
    },
  },
});

export const { setUserData, authenticate, updateUserName, logout } =
  authSlice.actions;
export default authSlice.reducer;
