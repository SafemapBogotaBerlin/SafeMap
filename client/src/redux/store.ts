import { configureStore } from '@reduxjs/toolkit';
import homeReducer from './home/index';
import authReducer from './session/index';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    auth: authReducer,
  },
});
export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
