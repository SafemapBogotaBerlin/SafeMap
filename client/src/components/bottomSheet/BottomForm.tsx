import React, { useRef } from 'react';
import { View, Animated } from 'react-native';
import ModalForm from '../addPointForm/ModalForm';
import GoProfileForm from '../goProfileForm/GoProfileForm';
import { useSelector } from 'react-redux';
import { RootState } from '../../redux/store';

const BottomForm = ({ fillType }: { fillType: string }) => {
  const slideAnimation = useRef(new Animated.Value(500)).current;
  const whatShouldBeOpened: string = useSelector(
    (state: RootState) => state.home.whatShouldBeOpened
  );

  return (
    <View>
      {whatShouldBeOpened === 'profile' ? <GoProfileForm /> : <ModalForm />}
    </View>
  );
};

export default BottomForm;
