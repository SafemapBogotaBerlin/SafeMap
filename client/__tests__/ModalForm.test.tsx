import React from 'react';
import renderer from 'react-test-renderer';
import BottomForm from '../src/components/navigation/Navigation';
import { Provider } from 'react-redux';
import { store } from '../src/redux/store';

describe('<Navigation/>', async () => {
  let component: any;

  beforeEach(() => {
    component = renderer.create(
      <Provider store={store}>
        <BottomForm fillType={''} />
      </Provider>
    );
  });

  it('renders  correctly', () => {
    const tree: string = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Spinner has one child', () => {
    const numberOfChildren = React.Children.count(
      component.root.props.children
    );
    expect(numberOfChildren).toBe(1);
  });

  it('Spinner component renders with correct color', () => {
    const tree = component.toJSON();

    const activityIndicator = tree.children.find(
      (child) => child.type === 'ActivityIndicator'
    );

    expect(activityIndicator.props.color).toBe('##0C5A1E');
  });
});
