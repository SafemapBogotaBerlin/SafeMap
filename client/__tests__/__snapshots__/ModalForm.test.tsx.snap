// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalForm /> renders  correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f8f8f8",
      "borderRadius": 20,
      "height": "70%",
      "margin": 10,
      "padding": 20,
      "shadowColor": "rgba(0, 0, 0, 0.1)",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 8,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderColor": "#ddd",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 3,
        "flexDirection": "row-reverse",
        "height": 50,
        "justifyContent": "space-between",
        "margin": 3,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 8,
        "paddingLeft": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#444",
            "fontSize": 18,
          },
          undefined,
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      numberOfLines={1}
      style={
        {
          "color": "#555",
          "flex": 1,
          "fontFamily": "Rubik_300Light",
          "fontSize": 16,
          "marginHorizontal": 8,
          "textAlign": "left",
        }
      }
    >
      Select event type
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderColor": "#ddd",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 3,
        "flexDirection": "row-reverse",
        "height": 50,
        "justifyContent": "space-between",
        "margin": 3,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 8,
        "paddingLeft": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#444",
            "fontSize": 18,
          },
          undefined,
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      numberOfLines={1}
      style={
        {
          "color": "#555",
          "flex": 1,
          "fontFamily": "Rubik_300Light",
          "fontSize": 16,
          "marginHorizontal": 8,
          "textAlign": "left",
        }
      }
    >
      When?
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#0C5A1E",
        "borderRadius": 10,
        "flex": 4,
        "justifyContent": "center",
        "margin": 3,
        "opacity": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#F2FAF6",
          "fontFamily": "Rubik_600SemiBold",
          "fontSize": 30,
          "fontWeight": "600",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
